File: src/dataset/javascript/acronym_exercism-ThomasZumsteg/gemini/GeminiFlash_acronym_v4.js
module.exports = {
    parse: parse
};

function parse(phrase) {
    return phrase.split(/\W+/)
        .map(word => {
            if (!word) return '';
            const firstChar = word.charAt(0).toUpperCase();
            const rest = word.slice(1);

            if (word.length > 1 && /^[A-Z]+$/.test(word)) {
                return firstChar;
            }

            let result = firstChar;
            for (let i = 0; i < rest.length; i++) {
                const char = rest[i];
                if (char >= 'A' && char <= 'Z') {
                    result += char;
                }
            }
            return result;
        })
        .join('');
}

File: src/dataset/Java/acronym_exercism-ThomasZumsteg/gemini/GeminiFlash_Acronym_v4.java
public class Acronym {
    public static String generate(String phrase) {
        String[] words = phrase.split("\\W+");
        StringBuilder acronym = new StringBuilder();
        for (String word : words) {
            if (word.isEmpty()) continue;
            acronym.append(Character.toUpperCase(word.charAt(0)));
            if (word.length() > 1) {
                for (int i = 1; i < word.length(); i++) {
                    char c = word.charAt(i);
                    if (Character.isUpperCase(c)) {
                        acronym.append(c);
                    }
                }
            }
        }
        return acronym.toString();
    }
}